### DISCUSSION TITLE
Support for xgen and codegen models from Salesforce

### DISCUSSION THREAD

**Discussion by RahulVivekNair** [ggml-org/llama.cpp#2137] [Ideas]
Salesforce has released weights for xgen and well as codegen which shows some impressive human eval numbers and the best part is that they are based on the llama architecture and are pre trained on upto 8k context length. Any chance that it can be run by llama.cpp considering it is the same architecture?

**Comment by tmostak** [comment]
I'd second this, finding that fine-tunes of the codegen 2.5 7B model are getting better results than Llama V2 13B on the same dataset.

**Comment by AayushSameerShah** [comment]
Hey!!!!
Looking for the same, I am willing to convert this: https://huggingface.co/NumbersStation/nsql-2B model into GGUF - which infact based on the CodeGen model.

Any luck with this?
